services/network/virtualnetwork/virtualnetwork.go

```diff
@@ -136,7 +136,13 @@ func getWssdNetworkSubnets(subnets *[]network.Subnet) (wssdsubnets []*wssdcloudn
 		}
 
 		if subnet.NetworkSecurityGroup != nil {
-			wssdsubnet.NetworkSecurityGroup = *subnet.NetworkSecurityGroup.ID
+			wssdsubnet.NetworkSecurityGroup = &wssdcommonproto.ResourceReference{
+				Name: *subnet.NetworkSecurityGroup.ID,
+			}
+
+			if location != nil {
+				wssdsubnet.NetworkSecurityGroup.Location = *location
```

Comment:
We should error out if `location` is nil, right? Or `nil` location is being handled in the cloud gent code?

Reply:
removed!